pipeline/alignments/generate-alignment-merged.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+##
+# Generates whitespace tokenized alignments jointly on original and back-translated corpus to work with OpusTrainer Tags
+#
+# It is used in the teacher model after back-translations
+#
+# OpusTrainer does not support SentencePiece tokenized alignments as input.
+# It will remap the whitespace tokenized alignments to match SentencePiece tokenization before passing to Marian.
+#
+
+set -x
+set -euo pipefail
+
+echo "###### Generating alignments"
+[[ -z "${BIN}" ]] && echo "BIN is empty"
+[[ -z "${SRC}" ]] && echo "SRC is empty"
+[[ -z "${TRG}" ]] && echo "TRG is empty"
+
+original_prefix=$1
+backtranslated_prefix=$2
+output_original_prefix=$3
+output_backtranslated_prefix=$4
+
+COMPRESSION_CMD="${COMPRESSION_CMD:-pigz}"
+ARTIFACT_EXT="${ARTIFACT_EXT:-gz}"
+
+cd "$(dirname "${0}")"
+
+output_dir=$(dirname "${output_original_prefix}")
+mkdir -p "${output_dir}"
+dir="${output_dir}/tmp"
+mkdir -p "${dir}"
+
+# train alignments on a merged corpus because fast_align is a statistical tool that benefits from a bigger corpus
+# and might not be accurate on a smaller one, for example when we have fewer back-translations
+
+# this is quite heavy on disk
+
+echo "### Decompressing original corpus to get its length"
+${COMPRESSION_CMD} -d "${original_prefix}.${SRC}.${ARTIFACT_EXT}"
```

Comment:
This feels very inefficient, and will lead to a ton of diskspace usage. In python you can make streaming iterators that you can pass around.